rllib/offline/estimators/direct_method.py

```diff
@@ -0,0 +1,159 @@
+    over episodes from the given batch. If the number of episodes in the
+    batch is less than `k` or `should_train` is set to False, yields an empty
+    list for train_episodes and all the episodes in test_episodes.
+
+    Args:
+        batch: A SampleBatch of episodes to split
+        k: Number of cross-validation splits
+        should_train: True by default. If False, yield [], [episodes].
+
+    Returns:
+        A tuple with two lists of SampleBatches (train_episodes, test_episodes)
+    """
+    episodes = batch.split_by_episode()
+    n_episodes = len(episodes)
+    if n_episodes < k or not should_train:
+        yield [], episodes
+        return
+    n_fold = n_episodes // k
+    for i in range(k):
+        train_episodes = episodes[: i * n_fold] + episodes[(i + 1) * n_fold :]
+        if i != k - 1:
+            test_episodes = episodes[i * n_fold : (i + 1) * n_fold]
+        else:
+            # Append remaining episodes onto the last test_episodes
+            test_episodes = episodes[i * n_fold :]
+        yield train_episodes, test_episodes
+    return
+
+
+class DirectMethod(OffPolicyEstimator):
+    """The Direct Method estimator.
+
+    q_model_type: Either "fqe" for Fitted Q-Evaluation or "qreg" for Q-Regression, or
+        a custom model that implements `estimate_q(states,actions)`
+        and `estimate_v(states, action_probs)`
+    framework: One of "tf|tf2|tfe|torch", currently only "torch" is supported
+    k: k-fold cross validation for training model and evaluating OPE
+    q_model_kwargs: Optional arguments for the specified Q model
+
+    DM estimator described in https://arxiv.org/pdf/1511.03722.pdf"""
+
+    @override(OffPolicyEstimator)
+    def __init__(
+        self,
+        name: str,
+        policy: Policy,
+        gamma: float,
+        q_model_type: str = "fqe",
+        framework: str = "torch",
+        k: int = 5,
+        **q_model_kwargs,
+    ):
+        super().__init__(name, policy, gamma)
+        # TODO (rohan): Add support for continuous action spaces
+        assert isinstance(
+            policy.action_space, Discrete
+        ), "DM Estimator only supports discrete action spaces!"
+        assert (
+            policy.config["batch_mode"] == "complete_episodes"
+        ), "DM Estimator only supports batch_mode=`complete_episodes`"
+        assert framework == "torch", "DM estimator only supports `framework`=`torch`"
+
+        # TODO (rohan): Add support for TF!
+        if framework == "torch":
+            if q_model_type == "qreg":
+                model_cls = QRegTorchModel
+            elif q_model_type == "fqe":
+                model_cls = FQETorchModel
+            else:
+                assert hasattr(
+                    q_model_type, "estimate_q"
+                ), "q_model_type must implement `estimate_q`!"
+                assert hasattr(
+                    q_model_type, "estimate_v"
+                ), "q_model_type must implement `estimate_v`!"
+
```

Comment:
Add an error here explaining that this estimator type is not available if `config.framework_str != "torch"`.

Reply:
Is this different from the `assert framework==`torch``? Technically, I think you can actually run this with policy as any PolicyV2.